File: valid_parentheses.py
def isValid(s):
    stack = []
    bracket = {')': '(', '}': '{', ']': '['}
    for ch in s:
        if ch in bracket:
            try:
                if bracket[ch] == stack[-1]:
                    stack.pop()
                    continue
            except:
                return False
        stack.append(ch)
    if len(stack) == 0:
        return True
    return False


print(isValid(']'))

File: valid_parentheses.cpp
#include<iostream>
#include<string>
#include<stack>
#include<unordered_map>
using namespace std;


bool isValid(string s) {
    unordered_map<char, char> bracket = {
        {')', '('},
        {'}', '{'},
        {']', '['}
    };
    unordered_map<char, char>::iterator iter;
    stack<char> stack;

    for(char ch : s){
        iter = bracket.find(ch);
        if(iter != bracket.end()){
            if(!stack.empty()){
                if(bracket[ch] == stack.top()){
                    stack.pop();
                    continue;
                }
            }
        }
        stack.push(ch);
    }

    if(stack.size() == 0) return true;
    return false;
}


int main(){
    cout<<isValid("[[[]]][")<<endl;

}
